From: Michal Hocko <mhocko@suse.com>
Subject: kabi: hwpoison, memory_hotplug: allow hwpoisoned pages to be offlined
Patch-mainline: never, kabi fixup
References: bnc#1116336

patches.fixes/0001-hwpoison-memory_hotplug-allow-hwpoisoned-pages-to-be.patch
has added an include and that has made some types visible and that confuses
kabi checker so hide the include from it.

KABI: symbol __online_page_free(vmlinux) changed crc from 0x081714c4 to 0xa9864d14
KABI: symbol __online_page_increment_counters(vmlinux) changed crc from 0xe5a58a95 to 0x5f4fd462
KABI: symbol __online_page_set_limits(vmlinux) changed crc from 0x9dde2b0a to 0x776b91e6
KABI: symbol __remove_pages(vmlinux) changed crc from 0xdbbbb628 to 0x0e240222
KABI: symbol restore_online_page_callback(vmlinux) changed crc from 0x73de2a3a to 0x6d487d01
KABI: symbol set_online_page_callback(vmlinux) changed crc from 0x91c09d5b to 0xdece3e4c
KABI: aborting due to kabi changes.
Detailed kabi warnings:
mm/memory_hotplug.c:653: warning: __remove_pages: modversion changed because of changes in struct anon_vma (became defined)
mm/memory_hotplug.c:673: warning: set_online_page_callback: modversion changed because of changes in struct anon_vma (became defined)
mm/memory_hotplug.c:692: warning: restore_online_page_callback: modversion changed because of changes in struct anon_vma (became defined)
mm/memory_hotplug.c:697: warning: __online_page_set_limits: modversion changed because of changes in struct anon_vma (became defined)
mm/memory_hotplug.c:703: warning: __online_page_increment_counters: modversion changed because of changes in struct anon_vma (became defined)
mm/memory_hotplug.c:709: warning: __online_page_free: modversion changed because of changes in struct anon_vma (became defined)

Signed-off-by: Michal Hocko <mhocko@suse.com>

---
 mm/memory_hotplug.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/mm/memory_hotplug.c
+++ b/mm/memory_hotplug.c
@@ -34,7 +34,9 @@
 #include <linux/memblock.h>
 #include <linux/bootmem.h>
 #include <linux/compaction.h>
+#ifndef __GENKSYMS__
 #include <linux/rmap.h>
+#endif
 
 #include <asm/tlbflush.h>
 
